### PR TITLE
Add sync-folders extension

### PR DESCRIPTION
**Pull request by gfazioli**
## Description

This mod provides a command to sync folders with the `rsync` command. It provides a simple way to sync folders between two locations. Take a look at the scenarios below to get a feel for how you might use the extension.

## Features

- **Sync Folders**: Sync folders between two locations.
- **Sync Finder Selection**: Sync the selected folders in Finder.
- **Create Sync Folders**: Create a new Sync folders preset.
- **Edit Sync Folders**: Edit an existing Sync folders preset.
- **My Sync Folders**: List all the Sync Folders presets.
- **Menu bar commands**: Quickly access the list of Sync Folders presets from the menu bar.

## Screencast

sync-folders-1
sync-folders-2
sync-folders-3
sync-folders-4
sync-folders-5
sync-folders-6
sync-folders-7

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/sync-folders/src/components/SyncFoldersForm.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            style={Action.Style.Regular}
+            onSubmit={handleSubmit}
+          />
+        </ActionPanel>
+      }
+    >
+      <Form.TextField title="Name" placeholder="Enter a name" {...itemProps.name} />
+
+      <Form.FilePicker
+        title="Select source folder"
```
Comment (pernielsentikaer) [inline extensions/sync-folders/src/components/SyncFoldersForm.tsx:92]:
```suggestion
        title="Select Source Folder"
```

File: extensions/sync-folders/src/components/SyncFoldersForm.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        allowMultipleSelection={false}
+        canChooseDirectories
+        canChooseFiles={false}
+        {...itemProps.source_folder}
+      />
+      <Form.FilePicker
+        title="Select dest folder"
```
Comment (pernielsentikaer) [inline extensions/sync-folders/src/components/SyncFoldersForm.tsx:99]:
```suggestion
        title="Select Dest Folder"
```

File: extensions/sync-folders/src/my-sync-folders.tsx [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+          return (
+            <List.Item
+              key={id}
+              icon={delete_dest ? "folders-orange.svg" : "folders-green.svg"}
+              title={name as string}
+              actions={
+                <ActionPanel title="Start Sync">
+                  <Action.SubmitForm
+                    style={Action.Style.Destructive}
```
Comment (pernielsentikaer) [inline extensions/sync-folders/src/my-sync-folders.tsx:160]: It's not a destructive action 🙂

File: extensions/sync-folders/src/sync-folders.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    return false;
+  }
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm
+            title="Start Sync"
+            style={Action.Style.Destructive}
```
Comment (pernielsentikaer) [inline extensions/sync-folders/src/sync-folders.tsx:71]: It's not a destructive action 🙂

File: extensions/sync-folders/src/sync-folders.tsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        </ActionPanel>
+      }
+    >
+      <Form.Description
+        title="Sync Folders"
+        text="Here you can sync two folders. The source folder will be copied to the destination folder. If the delete option is enabled, any files in the destination folder that are not in the source folder will be deleted."
+      />
+
+      <Form.FilePicker
+        title="Select source folder"
```
Comment (pernielsentikaer) [inline extensions/sync-folders/src/sync-folders.tsx:109]:
```suggestion
        title="Select Source Folder"
```

File: extensions/sync-folders/src/sync-folders.tsx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+        onBlur={(event) => {
+          if (event.target.value?.length == 0) {
+            setSourceFolderError("The field should't be empty!");
+          } else {
+            dropSourceFolderErrorIfNeeded();
+          }
+        }}
+      />
+      <Form.FilePicker
+        title="Select dest folder"
```
Comment (pernielsentikaer) [inline extensions/sync-folders/src/sync-folders.tsx:125]:
```suggestion
        title="Select Dest Folder"
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by gfazioli** [comment]
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

everything should fixed 👍

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**